core/src/main/java/tech/tablesaw/filtering/deferred/And.java

```diff
@@ -0,0 +1,30 @@
+package tech.tablesaw.filtering.deferred;
+
+import com.google.common.base.Preconditions;
+import tech.tablesaw.api.Table;
+import tech.tablesaw.selection.Selection;
+
+import java.util.function.Function;
+
+public class And implements Function<Table, Selection> {
+
+  private Function<Table, Selection>[] arguments;
+
+  public And(Function<Table, Selection>... arguments) {
+    Preconditions.checkArgument(
+        arguments != null && arguments.length > 0,
+        "The arguments to And must be a non-null array of length 1 or greater");
+    this.arguments = arguments;
+  }
+
+  @Override
+  public Selection apply(Table table) {
+    Selection result = arguments[0].apply(table);
+    if (arguments.length > 1) {
```

Comment:
if arguments length  = 1, i think it will NPE on arguments[i]

perhaps this could be written better, but i think it's needed as is.

Reply:
For the arguments length = 1 case since the loop starts at 1,  i < arguments.length is always false.

Try running this test if you do not believe me.


```java
  @Test
  public void testIndex() {

    String[] args = new String[]{"arg"};

    for (int i = 1; i < args.length; i++) {
        throw new RuntimeException();
    }
  }
```

Reply:
fixed